### PR TITLE
feat: [SRM-12425]: Adding API to fetch all AWS Prometheus workspaces

### PR DESCRIPTION
**Pull request by dhruv-harness**
## Checklist
- [ ] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/38941)

### CODE REVIEW

File: 948-cv-nextgen-beans/src/main/java/io/harness/cvng/beans/DataSourceType.java [hunk 1]
```diff
@@ -28,9 +28,9 @@ public enum DataSourceType {
   DYNATRACE("Dynatrace", VerificationType.TIME_SERIES, "prometheus"),
   CUSTOM_HEALTH_METRIC("CustomHealthMetric", VerificationType.TIME_SERIES, "prometheus"),
   CUSTOM_HEALTH_LOG("CustomHealthLog", VerificationType.LOG, "splunk"),
-
   ELASTICSEARCH("ElasticSearch", VerificationType.LOG, "splunk"),
-  CLOUDWATCH_METRICS("CloudWatchMetrics", VerificationType.TIME_SERIES, "appdynamics");
+  CLOUDWATCH_METRICS("CloudWatchMetrics", VerificationType.TIME_SERIES, "appdynamics"),
```
Comment (n00bitax) [inline 948-cv-nextgen-beans/src/main/java/io/harness/cvng/beans/DataSourceType.java:32]: Just out of curiosity a layman question, what does cloudwatch have to do with appdynamics ?

Comment (dhruv-harness) [reply 948-cv-nextgen-beans/src/main/java/io/harness/cvng/beans/DataSourceType.java:32]: They have similar data formats. That property is used for some demo purposes only.

Comment (n00bitax) [reply 948-cv-nextgen-beans/src/main/java/io/harness/cvng/beans/DataSourceType.java:32]: ideally, we shouldn't put demo code throughout our codebase, that's for another conversation. closing this thread,

**Review by AbhijithHarness — APPROVED** [review]
> Reviewed 8 of 11 files at r1, 6 of 8 files at r2, 1 of 2 files at r3, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38941)* status: :shipit: complete! ✓

**Comment by dhruv-harness** [comment]
trigger codebasehash

**Pull request merged**